csharp/src/Ice/Internal/LoggerAdminI.cs

```diff
@@ -76,30 +76,36 @@ public override void
 
         try
         {
-            remoteLogger.initAsync(_logger.getPrefix(), initLogMessages.ToArray()).ContinueWith(
-                (t) =>
-                {
-                    try
-                    {
-                        t.Wait();
-                        if (_traceLevel > 1)
-                        {
-                            _logger.trace(_traceCategory, "init on `" + remoteLogger.ToString()
-                                          + "' completed successfully");
-                        }
-                    }
-                    catch (AggregateException ae)
-                    {
-                        Debug.Assert(ae.InnerException is Ice.LocalException);
-                        deadRemoteLogger(remoteLogger, _logger, (Ice.LocalException)ae.InnerException, "init");
-                    }
-                });
+            _ = performInitAsync(remoteLogger, initLogMessages.ToArray());
```

Comment:
Since performInitAsync is "async", it won't throw any exception synchronously. So the try/catch block in superfluous.

I think you could also remove the local function parameters since there is no loop or ambiguity.

Reply:
Why not just do Task.Run for these?